frame/support/src/traits.rs

```diff
@@ -297,44 +297,72 @@ mod test_impl_filter_stack {
 	}
 }
 
+/// Error that can be returned by our impl of `StoredMap`.
+pub enum StoredMapError {
+	/// Attempt to create map value when it is a consumer and there are no providers in place.
+	NoProviders,
+	/// Attempt to anull/remove value when it is the last provider and there is still at
+	/// least one consumer left.
+	ConsumerRemaining,
+}
+
 /// An abstraction of a value stored within storage, but possibly as part of a larger composite
 /// item.
-pub trait StoredMap<K, T> {
+pub trait StoredMap<K, T: Default> {
```

Comment:
Is it really necessary to have `T: Default`?

Reply:
I could not make it work without.

Reply:
afaik, change storage from non option type to option type does not require migration. Option wrapping is handled by the decl_storage macro.

Reply:
+1 that option should not change the underlying storage

Reply:
The option would not be for a storage item itself, but rather an inner component of a storage item's structure. Concretely, it would be the `data` field of the `frame_system::AccountInfo` (i.e. `T::AccountData`) which would need to become an `Option`, which would require a migration.